hoomd/mpcd/Integrator.cc

```diff
@@ -174,6 +177,6 @@ void mpcd::detail::export_Integrator(pybind11::module& m)
                       &mpcd::Integrator::getStreamingMethod,
                       &mpcd::Integrator::setStreamingMethod)
         .def_property("solvent_sorter", &mpcd::Integrator::getSorter, &mpcd::Integrator::setSorter)
-        .def_property("filler", &mpcd::Integrator::getFiller, &mpcd::Integrator::setFiller);
+        .def_property_readonly("fillers", &mpcd::Integrator::getFillers);
```

Comment:
By default, pybind11 returns `std::vector<>` as a copy in a Python list. To enable the synced list infrastructure to update the vector in place:
1) Add
    ```
    #include <pybind11/stl_bind.h>
    PYBIND11_MAKE_OPAQUE(std::vector<std::shared_ptr<mpcd::VirtualParticleFiller>);
    ```
    to the top of the file.
2. Add 
    ```
    pybind11::bind_vector<std::vector<std::shared_ptr<ForceCompute>>>(m, "VirtualParticleFillerList");
    ```
   in the `export_Integrator` function. See [the pybind11 docs](https://pybind11.readthedocs.io/en/stable/advanced/cast/stl.html#making-opaque-types) for details.

Without these changes, you should find that `integrator.fillers = [filler1, filler1]` does not change the items in the C++ `std::vector`.

Reply:
Good catch! I have had to do this a bunch in another project but didn't think about it here. This should be fixed, and I added a unit test for something like your example.